src/routes/transactions/mappers/module-transactions/module-transaction-details.mapper.ts

```diff
@@ -0,0 +1,80 @@
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import { AddressInfoHelper } from '../../../common/address-info/address-info.helper';
+import {
+  MODULE_TRANSACTION_PREFIX,
+  TRANSACTION_ID_SEPARATOR,
+} from '../../constants';
+import { TransactionData } from '../../entities/transaction-data.entity';
+import { ModuleExecutionDetails } from '../../entities/transaction-details/module-execution-details.entity';
+import { TransactionDetails } from '../../entities/transaction-details/transaction-details.entity';
+import { TransactionDataMapper } from '../common/transaction-data.mapper';
+import { MultisigTransactionInfoMapper } from '../common/transaction-info.mapper';
+import { ModuleTransactionStatusMapper } from './module-transaction-status.mapper';
+
+@Injectable()
+export class ModuleTransactionDetailsMapper {
+  constructor(
+    private readonly addressInfoHelper: AddressInfoHelper,
+    private readonly statusMapper: ModuleTransactionStatusMapper,
+    private readonly transactionInfoMapper: MultisigTransactionInfoMapper,
+    private readonly transactionDataMapper: TransactionDataMapper,
+  ) {}
+
+  async mapDetails(
+    chainId: string,
+    transaction: ModuleTransaction,
+    safe: Safe,
+  ): Promise<TransactionDetails> {
+    const moduleAddress = await this.addressInfoHelper.getOrDefault(
+      chainId,
+      transaction.module,
+    );
+
+    return {
+      safeAddress: safe.address,
+      txId: `${MODULE_TRANSACTION_PREFIX}${TRANSACTION_ID_SEPARATOR}${safe.address}${TRANSACTION_ID_SEPARATOR}${transaction.moduleTransactionId}`,
+      executedAt: transaction.executionDate?.getTime() ?? null,
+      txStatus: this.statusMapper.mapTransactionStatus(transaction),
+      txInfo: await this.transactionInfoMapper.mapTransactionInfo(
+        chainId,
+        transaction,
+        safe,
+      ),
+      txData: await this.mapTransactionData(chainId, transaction),
+      txHash: transaction.transactionHash,
+      detailedExecutionInfo: new ModuleExecutionDetails(moduleAddress),
+      safeAppInfo: null,
+    };
+  }
+
+  private async mapTransactionData(
+    chainId: string,
+    transaction: ModuleTransaction,
+  ): Promise<TransactionData> {
+    const addressInfoIndex =
+      await this.transactionDataMapper.buildAddressInfoIndex(
+        chainId,
+        transaction.dataDecoded,
+      );
+    const trustedDelegateCallTarget =
+      await this.transactionDataMapper.isTrustedDelegateCall(
+        chainId,
+        transaction.operation,
+        transaction.to,
+        transaction.dataDecoded,
+      );
+
+    return {
+      to: await this.addressInfoHelper.getOrDefault(chainId, transaction.to),
+      value: transaction.value,
+      hexData: transaction.data,
+      dataDecoded: transaction.dataDecoded,
+      operation: transaction.operation,
+      addressInfoIndex: isEmpty(addressInfoIndex) ? null : addressInfoIndex,
```

Comment:
Yes, in fact, I was thinking in opening an issue for this because the following ID is giving me differnent `addressInfoIndex` for Nest and Rust implementations: `multisig_0x95ba94c6a2Dc8521754e7B8f4C1dFDBC9A271cA5_0x7e38a9f4c82776b3ad52c49011ed562041afa9b835687f6370f64bffe54d674c`

This was implemented for another endpoint (`/preview`), I verified it worked but for some reason I forgot to include the tests in the code 🤦🏻‍♂️. In you agree I can open the a bug for the result divergence, and also include the tests there.

Reply:
Totally fine! Let's keep an issue open to track it 👍

Reply:
I've created the issue: https://github.com/safe-global/safe-client-gateway-nest/issues/367 🙂

Reply:
> @fmrsabino @iamacook Just to make it clear: I was wrong when I said above that we don't have tests around `addressInfoIndex`, we do have:
> 
> https://github.com/safe-global/safe-client-gateway-nest/blob/5e9b5ace7de1f39098bdfb3033b49df019ec30c5/src/routes/transactions/mappers/common/transaction-data.mapper.spec.ts#L103

My comment was not around not having tests for the address info index. It was about having a test when `addressInfoIndex` is not empty for `module-transaction-details.mapper`:

> I don't think we have a test that tests when `addressInfoIndex` is not empty right?. Can be tackled on a separate PR.